runtime/jcl/common/system.c

```diff
@@ -356,11 +360,22 @@ jobject getPropertyList(JNIEnv *env)
 		}
 		strings[propIndex++] = !usernameAlloc || result ? "unknown" : usernameAlloc;
 	}
-
 #undef USERNAME_LENGTH
 
+	/* Set the maximum direct byte buffer allocation amount */
+
+	heapSize = javaVM->memoryManagerFunctions->j9gc_get_maximum_heap_size(javaVM);
+	/* allow up to 7/8 of the heap to be direct byte buffers */
+	directMemoryMax = heapSize - (heapSize / 8);
+	/* this property is consumed by java.nio.Bits */
+	strings[propIndex] = "sun.nio.MaxDirectMemorySize";
+	propIndex += 1;
+	j9str_printf(PORTLIB, maxDirectMemAlloc, sizeof(maxDirectMemAlloc)), "%zu", directMemoryMax);
+	strings[propIndex] = maxDirectMemAlloc;
+	propIndex += 1;
```

Comment:
This can also be set by `-XX:MaxDirectMemorySize` command line option.  Search our code for `VMOPT_XXMAXDIRECTMEMORYSIZEEQUALS`.

Setting it here would override the user's setting.

Reply:
See code in https://github.com/eclipse/openj9/blob/f1ade0f8f3baa4be78b00afa7ad499e67e196b30/runtime/util/vmargs.c#L1587-L1609 for where we set `sun.nio.MaxDirectMemorySize`

Reply:
...and that code is executed before `Java_java_lang_System_getPropertyList`. It also run before the GC has parsed the memory settings. so I can't just call `j9gc_get_maximum_heap_size()`.

Reply:
The separation between the jvm and vm libraries is largely pointless today, so if moving the option parsing further in fixes things, I'd suggest that.